### PR TITLE
Feature: Cropping Alignment Algorithms

### LINKED ISSUE #17: More alignment algorithm.
As title mentioned, add more cropping area alignment algorithm.

### PR DESCRIPTION
**Pull request by h-alice**
Now we implemented more cropping algorithms. 🚀

### CODE REVIEW

File: operation/image_cropping.go [hunk 1]
```diff
@@ -8,6 +8,15 @@ import (
 	"golang.org/x/image/draw"
 )
 
+// Define constants.
+const (
```
Comment (NatLee) [inline operation/image_cropping.go:12]: Good constant :D

File: operation/image_cropping.go [hunk 2]
```diff
@@ -62,13 +71,70 @@ func TopLeftAlignment(original_image_boundary image.Rectangle, cropped_width, cr
 	return image.Rect(crop_start.X, crop_start.Y, crop_end.X, crop_end.Y)
 }
 
+func BottomLeftAlignment(original_image_boundary image.Rectangle, cropped_width, cropped_height int) image.Rectangle {
```
Comment (NatLee) [inline operation/image_cropping.go:74]: Here share another insight with you. :D

```go
type Alignment struct {
	Horizontal string // "left" or "right"
	Vertical   string // "top" or "bottom"
}

func CropImage(boundary image.Rectangle, width, height int, align Alignment) image.Rectangle {
	startX, startY := calculateStartPoint(boundary, width, height, align)
	endX, endY := startX+width, startY+height
	return image.Rect(startX, startY, endX, endY)
}

func calculateStartPoint(boundary image.Rectangle, width, height int, align Alignment) (int, int) {
	var startX, startY int

	switch align.Horizontal {
	case "right":
		startX = boundary.Dx() - width
	case "left":
		startX = 0
	}

	switch align.Vertical {
	case "bottom":
		startY = boundary.Dy() - height
	case "top":
		startY = 0
	}

	return startX, startY
}

```

File: operation/image_resize.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func resizeAlgotithm(algo string) (algorithm draw.Interpolator) {
 		algorithm = draw.NearestNeighbor
 	case "catmullrom": // Default algorithm, best quality.
 		algorithm = draw.CatmullRom
-	case "approxbiLinear": // Somehow decent quality.
+	case "approxbilinear": // Somehow decent quality.
```
Comment (NatLee) [inline operation/image_resize.go:21]: Wow! It's in lower-case. :O

**Review by NatLee — APPROVED** [review]
Great news! No problems were found with this push. 👍

File: operation/image_cropping.go (hunk 1 above)
Comment (h-alice) [reply operation/image_cropping.go:12]: So sad that Go has no `Enum`. : (

**Pull request merged**